lib/modules/dosomething/dosomething_taxonomy/dosomething_taxonomy.module

```diff
@@ -110,15 +110,21 @@ function dosomething_taxonomy_preprocess_taxonomy_term(&$vars) {
 
 /**
  * Returns array of node nid's with the given Term $tid.
- * @deprecated Use taxonomy_select_nodes instead.
  *
  * @param int $tid
  *   A taxonomy term tid.
  *
  * @return array
  */
 function dosomething_taxonomy_get_campaigns($tid) {
-  return taxonomy_select_nodes($tid, FALSE);
+  $nids = taxonomy_select_nodes($tid, FALSE);
+  $campaigns = [];
+  foreach ($nids as $delta => $nid) {
+    if (node_load($nid)->type === 'campaign') {
```

Comment:
oh jes! good call @DFurnes Drupal does do some caching of nodes on it's own for speed, but yeah, `node_load_multiple` would be more ideal.